src/rust/rust-proto/src/client_macros.rs

```diff
@@ -29,34 +51,48 @@ macro_rules! execute_client_rpc {
         $rpc_name: ident,
         $proto_request_type: ty,
         $native_response_type: ty,
-        $opts: expr,
+        $rpc_config: expr,
     ) => {{
         {
-            let backoff = client_executor::strategy::FibonacciBackoff::from_millis(100)
-                .max_delay(Duration::from_millis(5000))
-                .map(client_executor::strategy::jitter);
-            let num_retries = 10;
+            // Bind the macro arguments to well typed things.
+            let rpc_config: $crate::client_macros::RpcConfig = $rpc_config;
+            let backoff_opts: $crate::client_macros::RpcBackoffConfig = rpc_config.backoff;
+            let proto_client = $self.proto_client.clone();
+            type ProtoRequestType = $proto_request_type;
+            type NativeResponseType = $native_response_type;
 
-            let proto_request = <$proto_request_type>::try_from($native_request)?;
+            // See docs on RpcBackoffConfig::default for a concrete example of this retry backoff timing.
+            let backoff = client_executor::strategy::FibonacciBackoff::from_millis(backoff_opts.initial_millis)
+                .max_delay(Duration::from_millis(backoff_opts.max_delay_millis))
+                .map(client_executor::strategy::jitter);
 
+            // Always retry if Unavailable, and optionally retry if
+            // specified by the RpcConfig.
             let executor_retry_condition = |status: &tonic::Status| {
-                // Always retry if Unavailable, and optionally retry if
-                // specified by the ExecuteClientRpcOptions.
-                status.code() == tonic::Code::Unavailable || ($opts.retry_predicate)(status)
+                status.code() == tonic::Code::Unavailable || (rpc_config.retry_predicate)(status)
             };
             let span = tracing::span::Span::current();
 
-            let proto_response = $self
+            // Convert the Native request - something in `rust-proto` - into a Protobuf request.
+            let proto_request: ProtoRequestType = ProtoRequestType::try_from($native_request)?;
+
+            // Execute the RPC, with retries.
+            let tonic_response: tonic::Response<_> = $self
                 .executor
                 .spawn_conditional(
-                    backoff.take(num_retries),
+                    backoff.take(backoff_opts.num_retries),
```

Comment:
in order for this to work in `async-cache` we need to be able to chart a path through this macro which _avoids_ specifying a retry count